hack/jenkins/test-flake-chart/report_flakes/report_flake.go

```diff
@@ -0,0 +1,258 @@
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	// read the file
+	data, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	var summary shortSummary
+	if err = json.Unmarshal(data, &summary); err != nil {
+		return nil, fmt.Errorf("failed to deserialize the file: %v", err)
+	}
+	return &summary, nil
+
+}
+
+// flakeRate downloads recent flake rates from GCS, and returns a map{env->map{testname->flake rate}}
+func flakeRate(ctx context.Context, client *storage.Client) (map[string]map[string]float64, error) {
+	btk := client.Bucket("minikube-flake-rate")
+	obj := btk.Object("flake_rates.csv")
+	reader, err := obj.NewReader(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read the flake rate file: %v", err)
+	}
+	// parse the csv file to the map
+	records, err := csv.NewReader(reader).ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("failed to read and parse the flake rate file: %v", err)
+	}
+	result := map[string]map[string]float64{}
+	for i := 1; i < len(records); i++ {
+		// for each line in csv we extract env, test name and flake rate
+		if len(records[i]) < 3 {
+			// the csv must have at least 2 columns
+			continue
+		}
+		env := records[i][0]
+		test := records[i][1]
+		flakeRate, err := strconv.ParseFloat(records[i][2], 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse the flake rate file at line %d: %v", i+1, err)
+		}
+		if _, ok := result[env]; !ok {
+			result[env] = make(map[string]float64, 0)
+		}
+		result[env][test] = flakeRate
+	}
+	return result, nil
+}
+
+func generateCommentMessage(summaries map[string]*shortSummary, flakeRates map[string]map[string]float64, pr, rootJob string) string {
+	type failedTest struct {
+		flakeRate float64
+		env       string
+		testName  string
+	}
+	// for each environment, we sort failed tests according to the flake rate of that test on master branch
+	envFailedTestList := map[string][]failedTest{}
+	for env, summary := range summaries {
+		failedTestList := []failedTest{}
+		for _, test := range summary.FailedTests {
+			// if we cannot find the test, we assign the flake rate as -1, meaning N/A
+			flakerate := unknown
+			if v, ok := flakeRates[env][test]; ok {
+				flakerate = v
+			}
+			failedTestList = append(failedTestList,
+				failedTest{
+					flakeRate: flakerate,
+					env:       env,
+					testName:  test,
+				})
+		}
+
+		sort.Slice(failedTestList, func(i, j int) bool {
+			return failedTestList[i].flakeRate < failedTestList[j].flakeRate
+		})
+		envFailedTestList[env] = failedTestList
+	}
+	// we convert the result into a 2d string slice representing a markdown table,
+	// whose each line represents a line of the table
+	table := [][]string{
+		// title of the table
+		{"Environment", "Test Name", "Flake Rate"},
+	}
+	// if an env has too much failures we will just skip it and print a message in the end
+	// if the failed tests have high flake rates(over 50% for all), it will also be skipped in the table
```

Comment:
```suggestion
	// if an env has too much failures we will just skip it and print a
	// message in the end if the failed tests have high flake rates (over
	// 50% for all), it will also be skipped in the table
```
Shortening long comments for better readability